fs/remote/blob.go

```diff
@@ -74,46 +69,39 @@ type blob struct {
 
 	size          int64
 	chunkSize     int64
-	cache         cache.BlobCache
 	lastCheck     time.Time
 	lastCheckMu   sync.Mutex
 	checkInterval time.Duration
 	fetchTimeout  time.Duration
 
-	fetchedRegionSet    regionSet
-	fetchedRegionSetMu  sync.Mutex
-	fetchedRegionGroup  singleflight.Group
-	fetchedRegionCopyMu sync.Mutex
+	fetchedRegionSet   regionSet
+	fetchedRegionSetMu sync.Mutex
 
 	resolver *Resolver
 
 	closed   bool
 	closedMu sync.Mutex
 }
 
-func makeBlob(fetcher fetcher, size int64, chunkSize int64,
-	blobCache cache.BlobCache, lastCheck time.Time, checkInterval time.Duration,
+func makeBlob(fetcher fetcher, size int64, chunkSize int64, lastCheck time.Time, checkInterval time.Duration,
 	r *Resolver, fetchTimeout time.Duration) *blob {
 	return &blob{
 		fetcher:       fetcher,
 		size:          size,
 		chunkSize:     chunkSize,
-		cache:         blobCache,
 		lastCheck:     lastCheck,
 		checkInterval: checkInterval,
 		resolver:      r,
 		fetchTimeout:  fetchTimeout,
 	}
 }
 
-func (b *blob) Close() error {
+func (b *blob) Close() {
```

Comment:
`Close() error` is a standard method from the [Closer interface](https://pkg.go.dev/io#Closer). The current implementation may not return an error, but it is generally useful to have `blob` implement *Closer* in its familiar form.  a) It might want to return an error again at some point. b) Gophers [expect](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) `Close()` to return an error. So have it do so makes the code more readable. c) It costs nothing to have it do so.

Reply:
Done. Please take a look at the latest revision